src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -59,6 +59,7 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
 
             // TODO: removing transactions from TX pool here seems to be a bad practice since they will
             // not come back if the block is ignored?
+            int blobsCounter = 0;
```

Comment:
Imo it's worth to change to sth like `remainingBlobsCount` = `4` and decrement after adding blob-type tx to the block